### ISSUE TITLE
Question: What is OMS-953 - fix for hidden password boxes?

### ISSUE THREAD

**Issue by BonfaceKilz** [operasoftware/browserjs#26]
I'm running opera mini, and I have a form which has hidden input elements that should be passed to the server in a post request as soon as the submit is clicked. However, after clicking the submit button, I'm getting the error:

```
ECMAscript errors

Opera has modified the JavaScript on <my-domain-name>(OMS-953 fix for hidden password boxes). See browser.js for details.
```

Whati does this error mean?

**Comment by psmyrak** [comment]
This is a compatibility patch that fixes errors that stem from webmaster assumptions that Mini servers would work as a local browser. Since the real browser runs on the server, certain events for hidden fields do not work as in a local browser. 

The original fix was to address issues reported regarding non working login forms on various sites. Thus the wording of the console debug message. Although since then the fix has become generic in nature.

**Comment by psmyrak** [comment]
Do you think our patch breaks your site?

**Comment by BonfaceKilz** [comment]
Yes it does, but I don't have a workaround for this. The behaviour is random. For older feature phones, the error above error is thrown. For newer phones, the error is not thrown. I'm appending a an `li` element that looks like this:
```
"<li class='element' id='div_" + span_id + "'><div id='txt_" + span_id + "' class='jack-bet " + span_id + "'><p><strong>" + match + " </strong><a href='#' id='remove_" + span_id + "' class='remove'><span class='icon-cancel15x'></span></a></p><p><span class='your-pick'>Your Pick: <b>" + match_value + " </b></span></p></div><input type='hidden' name='"+ match_id + "' value='"+ match_value +"' /></li>";
```
The above is appended by js to a the dom whose html looks like this:

**Comment by BonfaceKilz** [comment]
@psmyrak Thanks for responding. I thought this repo was stale since it ~ 3 years old.

**Comment by BonfaceKilz** [comment]
Are the compatibility patches public?

**Comment by psmyrak** [comment]
I do believe you are correct about the repo being stale, still your issue may be valid. I am tracking it within our internal system now.

**Comment by psmyrak** [comment]
I cannot reproduce your problem. The patch isn't touching `input[type=hidden]` at all, which make me believe that this is unrelated to the issue your observe. There is a tiny bug though that the patch log is posted despite the patch not doing anything, which is misleading.

**Comment by BonfaceKilz** [comment]
@psmyrak is it possible to update the repo. It would be really helpful to study how some things are processed :sweat_smile:

**Comment by BonfaceKilz** [comment]
Right now I'm operating by doing alot of blind guesses. I also noted that if you have a form, the inputs should not be part of a scrollable element. If you have a form with a long list of input elements, and the user can scroll down the input elements, Browserjs will throw the above error... and the rendered form will be cut view of the form. You also can't make a post request in this situation. I also noted that on low end phones(I tested on Techno N2S running Android 6.0), you have to click twice on button element for it to work as expected. This is not a problem on higher end phones(I used Redme 5plus to test this out).

**Comment by psmyrak** [comment]
The difference, you are seeing between the higher end devices and low end ones, may potentially stem from a various modes they operate in. On less powerful ones, you are likely running Extreme mode by default, while on a powerful one the High mode might be preselected. The difference is in that the Extreme mode has a browser engine running on server and device app is a display agent, which architecture has its limitations, while High is running locally just like most of other browsers. 

What client versions do you use for testing?

**Comment by BonfaceKilz** [comment]
Thanks for the info.

I'm using Opera v38.0.2254.134507 on the Redme 5 plus phone. On the feature phone, I'm running Opera v39.1.2254.136743. I hope this is what you mean by client version.

**Comment by psmyrak** [comment]
Yes, this is what I was looking for.
